Fixes and updates

Affordall opened the pull request:
- increase buildToolsVersion
- increase kotlin version to 1.4.0
- increase android gradle plugin version to 4.0.1
- correct dependencies
- fix sonar issues
- fix missing break in Medicine
- fix auto-close cursor

Commit:
fixed taxes for position

Commit:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

# Conflicts:
#	app/build.gradle

Commit:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

Commit:
Add BigDecimal readable comparison in Utils

Commit:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

Commit:
Add new field to Position

- Add new mandatory filed in Position - PaymentFeature as sealed class
- Make it parcelable
- Add new FFD payment types
- Add new FFD product types

Commit:
Add new field to Position

- Add new mandatory filed in Position - PaymentFeature
- Add mapper

Commit:
Add show list for NavigationAPi

Commit:
New implementation of PaymentFeature

Commit:
new position field

Commit:
counter-offer

Commit:
new PaymentFeature version

Commit:
new PaymentFeature version

Commit:
new version variable

Commit:
new name of version

Commit:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

# Conflicts:
#	src/main/java/ru/evotor/framework/core/action/datamapper/PaymentFeatureMapper.kt
#	src/main/java/ru/evotor/framework/core/action/datamapper/PositionMapper.java
#	src/main/java/ru/evotor/framework/navigation/NavigationApi.kt
#	src/main/java/ru/evotor/framework/payment/PaymentFeature.kt
#	src/main/java/ru/evotor/framework/receipt/Position.java

Commit:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

Affordall committed:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

Affordall committed:
Merge branch 'develop' of https://github.com/evotor/integration-library into develop

Affordall committed:
- increase buildToolsVersion
- increase kotlin version to 1.4.0
- increase android gradle plugin version to 4.0.1
- correct dependencies
- fix sonar issues
- fix missing break in Medicine
- fix auto-close cursor

xcam commented on src/main/java/ru/evotor/framework/inventory/InventoryApi.kt:
```diff
@@ -53,7 +46,7 @@ object InventoryApi {
                     arrayOf(productUuid),
                     null
-            )?.let {
+            )?.use {
```
а там toList курсор не закрывает?

Affordall replied:
нет

xcam commented on src/main/java/ru/evotor/framework/inventory/InventoryApi.kt:
```diff
@@ -65,41 +58,35 @@ object InventoryApi {
-                                        classificationCode = cursor.optString(cursor.getColumnIndex(ProductTable.ROW_CLASSIFICATION_CODE))
-                                )
-                            }
+                ?.use { cursor ->
```
catch потерял - теперь exception полетит наружу

Affordall replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/inventory/InventoryApi.kt:
```diff
@@ -135,17 +122,11 @@ object InventoryApi {
-                                result.add(createProductExtra(cursor))
-                            } while (cursor.moveToNext());
-                        }
-                    } catch (e: Exception) {
```
catch потерял - теперь exception полетит наружу

Affordall replied:
По-идее, оно и должно лететь наружу. Но я верну

xcam commented on src/main/java/ru/evotor/framework/inventory/InventoryApi.kt:
```diff
@@ -166,20 +147,13 @@ object InventoryApi {
-                        } else {
-                            return null
-                        }
-                    } catch (e: Exception) {
```
catch потерял - теперь exception полетит наружу

Affordall replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/receipt/ReceiptApi.kt:
```diff
@@ -263,11 +254,9 @@ object ReceiptApi {
                 type?.let { arrayOf(it.name) },
                 null
-        )?.let {
+        )?.use{
```
мы правда хотим отдавать наружу закрытый курсор?

Affordall replied:
да, но вернул)

xcam commented on src/main/java/ru/evotor/framework/receipt/ReceiptApi.kt:
```diff
@@ -280,7 +269,7 @@ object ReceiptApi {
     fun getFiscalReceipts(context: Context, receiptUuid: String): ru.evotor.query.Cursor<FiscalReceipt>? =
             context.contentResolver.query(FiscalReceiptContract.URI, null, null, arrayOf(receiptUuid), null)
-                    ?.let {
+                    ?.use {
```
мы правда хотим отдавать наружу закрытый курсор?

Affordall replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/users/UserApi.kt:
```diff
@@ -17,18 +16,12 @@ object UserApi {
-                            users.add(UserMapper.createUser(cursor))
-                        }
-                        return users
-                    } catch (e: Exception) {
```
catch потерялся

Affordall replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/users/UserApi.kt:
```diff
@@ -42,15 +35,9 @@ object UserApi {
-                        if (cursor.moveToFirst()) {
-                            return UserMapper.createUser(cursor)
-                        }
-                    } catch (e: Exception) {
```
catch потерялся

Affordall replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/users/UserApi.kt:
```diff
@@ -65,18 +52,12 @@ object UserApi {
-                            grants.add(UserMapper.createGrant(cursor))
-                        }
-                        return grants
-                    } catch (e: Exception) {
```
catch потерялся

Affordall replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/users/UserApi.kt:
```diff
@@ -88,24 +69,15 @@ object UserApi {
-                            grants.add(UserMapper.createGrant(cursor))
-                        }
-                        return grants
-                    } catch (e: Exception) {
```
catch потерялся

Affordall replied:
вернул

xcam reviewed (changes requested):
есть замечания

Affordall committed:
- fix code-review comments

Affordall committed:
- increased kotlin version

Affordall committed:
MRK-166

- Add fiscal tags
- Add default value to PreferentialMedicineType
- Add MedicineAttributeSubTable
- Mark as deprecated columns in PrintGroupSubTable
- Simplify method 'createPrintGroup' in ReceiptApi
